### DISCUSSION TITLE
Fehler bei modbus tcp Zähler als Master

### DISCUSSION THREAD

**Discussion by Pandur2001** [camueller/SmartApplianceEnabler#342] [Installation und Betrieb]
Ich habe bei mir einen modbus tcp Zähler in SAE konfiguriert. Dieser funktioniert soweit auch.
Wenn ich diesen Zähler allerdings als Master-Zähler konfiguriere, funktioniert er nicht mehr und ich bekomme folgende Fehlermeldungen:
`2022-09-17 12:54:30,094 ERROR [Timer-0] d.a.s.m.ModbusSlave [ModbusSlave.java:92] F-19760812-000000000030-00: Cannot connect to modbus zaehlerschrank
2022-09-17 12:55:30,094 ERROR [Timer-0] d.a.s.m.ModbusSlave [ModbusSlave.java:92] F-19760812-000000000030-00: Cannot connect to modbus zaehlerschrank`

Ich möchte mit dem Master-Zähler die PV-Anhebung realisieren wie hier beschrieben:
https://github.com/camueller/SmartApplianceEnabler/discussions/216

Welche Infos aus dem Log werden zur Analyse noch benötigt?
Setze Version 2.0.5 unter Docker ein.

Vielen Dank schon mal für die Hilfe und die tolle Software!
Grüße,
Sebastian

**Comment by camueller** [comment]
Kannst Du mir ein komplettes Log-File verfügbar machen (Dropbox, Google Drive, ...)

**Reply by Pandur2001** [reply]
Hier mal die letzten Logs:
https://www.dropbox.com/sh/rr836wudsu85kpi/AAB8Tp6Oqlour3U0QiYDysuFa?dl=0

**Comment by camueller** [comment]
Ich konnte den Fehler reproduzieren - die Initiliaisierung für Modbus-Zähler als Master-Zähler war fehlerhaft.
Über den nachfolgenden Link kannst Du eine 2.0.5 mit diversen Fixen herunterladen inkl. des Fixes für den Master-Zähler:

https://drive.google.com/file/d/13UHWBdYlipxPAwko5kxQToISg1w--hP-/view?usp=sharing

Bitte gib mir Rückmeldung, ob damit bei Dir Master/Slave-Zähler korrekt funktionieren. Ich habe bisher keine Rückmeldungen zu diesem, relativ neuen, Feature bekommen. Ich selbst werde es erst ab Beginn der Heiz-Saison nutzen, wobei ich allerdings keine Modbus-Zähler in diesem Kontext verwende.

**Reply by Pandur2001** [reply]
Super danke für den Fix! Ich versuche die neue Version in den nächsten Tagen mal zu testen und dann Feedback geben.
Heute werde ich es wohl leider nicht mehr schaffen.

**Reply by Pandur2001** [reply]
So, habe die war Datei getauscht und den Container neu gestartet.
Schaut bisher ganz gut aus! 
Modbus Zähler kann weiter ausgelesen werden, auch wenn ich die Option Masterzähler aktiviere.
Ich habe jetzt das Gerät mit dem Slave-Zähler konfiguriert. Mal schauen wie sich das die nächsten Tage verhält.

**Reply by realmuemmel** [reply]
Kurze Rückmeldung von mir, ich hatte in #216 das gleich Problem. Mit der Version oben funktioniert es jetzt wie es soll, danke!
